Source/VMware.vSphereDSC/Classes/VMHostBaseDSC.ps1

```diff
@@ -38,4 +50,97 @@ class VMHostBaseDSC : BaseDSC {
+        if ($this.RestartTimeout -le $timeSpan.Minutes) {
+            throw "Aborting the operation. VMHost $($this.Name) could not be restarted successfully in $($this.RestartTimeout) minutes."
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Ensures that the specified VMHost is in the desired state after successful restart operation.
+    #>
+    [void] EnsureVMHostIsInDesiredState($server, $desiredState) {
+        $sleepTimeInSeconds = 10
+        $elapsedTimeInSeconds = 0
+
+        while ($true) {
+            $this.EnsureRestartTimeoutIsNotReached($elapsedTimeInSeconds)
+
+            Start-Sleep -Seconds $sleepTimeInSeconds
+            $elapsedTimeInSeconds += $sleepTimeInSeconds
+
+            try {
+                if ($null -eq $server) {
+                    $this.ConnectVIServer()
+                }
+
+                $vmHost = $this.GetVMHost()
+                if ($vmHost.ConnectionState.ToString() -eq $desiredState) {
+                    break
+                }
+
+                Write-Verbose -Message "VMHost $($this.Name) is still not in $desiredState State."
+            }
+            catch {
+                <#
+                Here the message used in the try block is written again in the case when an exception is thrown
+                when retrieving the VMHost or establishing a Connection. This way the user still gets notified
+                that the VMHost is not in the Desired State.
+                #>
+                Write-Verbose -Message "VMHost $($this.Name) is still not in $desiredState State."
+            }
+        }
+
+        Write-Verbose -Message "VMHost $($this.Name) is successfully restarted and in $desiredState State."
+    }
+
+    <#
+    .DESCRIPTION
+
+    Restarts the specified VMHost so that the Update of the VMHost Configuration is successful.
+    #>
+    [void] RestartVMHost($vmHost) {
+        try {
+            Restart-VMHost -Server $this.Connection -VMHost $vmHost -Confirm:$false -ErrorAction Stop
+        }
+        catch {
+            throw "Cannot restart VMHost $($vmHost.Name). For more information: $($_.Exception.Message)"
+        }
+
+        <#
+        If the Connection is directly to a vCenter we do not need to establish a new connection so we pass the
+        current one to the method 'EnsureVMHostIsInCorrectState'. When the Connection is directly to an ESXi, after a successful
+        restart the ESXi is down so new Connection needs to be established to check the ESXi state. So the current
+        Connection is set to $null and then passed to the method 'EnsureVMHostIsInCorrectState'.
+        #>
+        if ($this.Connection.ProductLine -eq $this.vCenterProductId) {
+            $this.EnsureVMHostIsInDesiredState($this.Connection, $this.NotRespondingState)
+            $this.EnsureVMHostIsInDesiredState($this.Connection, $this.MaintenanceState)
+        }
+        else {
+            $this.Connection = $null
```

Comment:
Ah, here is the $null for the server param of EnsureVMHostIsInDesiredState. This means in the case $server will be always null. Probably the parameter shouldn't be $server but bool $requiresConnectVIServer